### PR TITLE
fix: karmada-operator install default version karmada

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

operator can judge which version karmada to install by operator itself's image version, inteaed of using fixed version in constant variable. (so, we will no longger need to manually submit PR like #3718)

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

how to test it:

```bash
#!/bin/bash

kind delete clusters --all; rm -rf ~/.karmada/; rm -rf ~/.kube/*.config; rm -rf /etc/karmada
cd path/to/karmada  # clone my submited code, and cd to your karmada repo path

hack/create-cluster.sh karmada-host ~/.kube/karmada-host.config
export KUBECONFIG=~/.kube/karmada-host.config

export VERSION="latest"
export REGISTRY="docker.io/karmada"
make image-karmada-operator GOOS="linux" --directory=.

kind load docker-image docker.io/karmada/karmada-operator:latest --name karmada-host
kind load docker-image registry.k8s.io/etcd:3.5.9-0 --name karmada-host
kind load docker-image registry.k8s.io/kube-apiserver:v1.25.4 --name karmada-host
kind load docker-image registry.k8s.io/kube-controller-manager:v1.25.4 --name karmada-host

helm install karmada-operator -n karmada-system  --create-namespace --dependency-update ./charts/karmada-operator --debug

kubectl apply -f operator/config/crds/

read -s -n1 -p "Press any key to continue ... "

cat >/tmp/karmada.yaml<<EOF
apiVersion: operator.karmada.io/v1alpha1
kind: Karmada
metadata:
  name: karmada
  namespace: karmada-system
EOF
kubectl apply -f karmada.yaml
```

test report:
image
image
image

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-operator`: The default installed version of Karmada now depends on the operator version.
```

### CODE REVIEW

**Comment by calvin0327** [comment]
@chaosi-zju Thanks, this is a good feature for operator. and this pr has some conflicts.

**Comment by chaosi-zju** [comment]
> @chaosi-zju Thanks, this is a good feature for operator. and this pr has some conflicts.

ok, I am fixing it~

**Comment by chaosi-zju** [comment]
I have rebased master again, ready to review ~

File: operator/config/samples/karmada.yaml [hunk 1]
```diff
@@ -31,19 +31,15 @@ spec:
       serviceSubnet: 10.96.0.0/12
     karmadaAggregatedAPIServer:
       imageRepository: docker.io/karmada/karmada-aggregated-apiserver
-      imageTag: v1.7.0
```
Comment (RainbowMango) [inline operator/config/samples/karmada.yaml:34]: We might not need to remove the version here.
I think even though `karmada-operator` can get the version, it is still recommended to declare the image version.

**Review by RainbowMango — COMMENTED** [review]
Generally looks good, just a nit.

File: operator/config/samples/karmada.yaml (hunk 1 above)
Comment (chaosi-zju) [reply operator/config/samples/karmada.yaml]: done

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**